### PR TITLE
Use built-in alertmanager config if no fallback config is provided

### LINKED ISSUE #2772: Provide a built-in Alertmanager default (empty) config
I'm seeing several people trying out Mimir Alertmanager being confused by the need of having to specify a per-tenant Alertmanager configuration in order to get it work. Because of this, in the past we've added `-alertmanager.configs.fallback` which allows you to specify the path to a local `.yaml` file containing a default Alertmanager configuration to use for any tenant for which hasn't been specified a configuration yet. However, most people is not aware of that until they start digging into "why Alertmanager API/UI doesn't work".

**Question**: what if we provide a built-it default Alertmanager config (which you can still override via `-alertmanager.configs.fallback`) to let Alertmanager working with an empty config out of the box?

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
#### Which issue(s) this PR fixes or relates to

Fixes #2772 

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/operators-guide/architecture/components/alertmanager.md
Comment (osg-grafana) [inline docs/sources/operators-guide/architecture/components/alertmanager.md:63]:
```suggestion
You can overwrite the default fallback configuration via the `-alertmanager.configs.fallback` command-line flag.
```

Comment (osg-grafana) [reply docs/sources/operators-guide/architecture/components/alertmanager.md:63]: Rationale: passive to active voice

File: CHANGELOG.md
Comment (JohnnyQQQQ) [reply CHANGELOG.md]: Changed 👍

**Comment by replay** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

File: CHANGELOG.md
Comment (stevesg) [inline CHANGELOG.md:10]: [nit] This didn't read cleanly to me, tried to make it a bit clearer

```suggestion
* [CHANGE] Alertmanager: added default configuration for `-alertmanager.configs.fallback`. Allows tenants to send alerts without first uploading an Alertmanager configuration. #2772
```

File: pkg/alertmanager/multitenant.go
Comment (stevesg) [inline pkg/alertmanager/multitenant.go:894]: [nit] This was probably just leftover from the last change

```suggestion
	// Upload an empty config so that the Alertmanager is no de-activated in the next poll
```

**Review by stevesg — COMMENTED** [review]
Change looks good conceptually. I spoke to @JohnnyQQQQ privately and said that we should move the change for whether or not to upload the default configuration into a separate PR, as I think it has some implications which warrant discussion.

I was thinking that we should add a functional test (either in `multitenant_test.go` or as an integraiton test), to the scenario where we do not set a fallback configuration file, but still attempt to use an alertmanager before it is configured. But I noticed this test already exists [here](https://github.com/grafana/mimir/blob/4141848b47c5c599c8a07d1477f0e882415aab0d/pkg/alertmanager/multitenant_test.go#L945-L957), which in theory should now fail, but doesn't seem to?

The one thing I'm missing

**Comment by JohnnyQQQQ** [comment]
There is also one in the [integration folder](https://github.com/grafana/mimir/blob/main/integration/alertmanager_test.go#L188-L190), which is not failing. I'm investigating it.

**Comment by JohnnyQQQQ** [comment]
The thing was that the mock that we are using in the test was built manually and not using the constructor, thus no fallback config was computed. I fixed that [here](https://github.com/grafana/mimir/pull/3541/files#diff-5252308a4554ebe5af672884ce76a033819a51531c357f3e9c0c36eda517be1cR105-R109), and now everything works as expected. The tests were failing and I rewrote them to reflect those changes and check for the default config instead of a 412 after a deletion. For the frontend tests the expected behavior is a redirect instead of a 412 after deletion, as the fallback config is used.

**Review by stevesg — APPROVED** [review]
LGTM, I think this is fine to merge once @grafana/docs-metrics approves the docs changes.

File: pkg/alertmanager/multitenant_test.go
Comment (osg-grafana) [inline pkg/alertmanager/multitenant_test.go:2178]:
```suggestion
	// If a fallback configuration file is set, it returns its content.
```

File: pkg/alertmanager/multitenant_test.go
Comment (osg-grafana) [inline pkg/alertmanager/multitenant_test.go:2171]:
```suggestion
	// If no fallback configration is set, it returns a valid empty configuration.
```

Comment (osg-grafana) [reply pkg/alertmanager/multitenant_test.go:2171]: To remove ambiguity and raise the tone of confidence, reword instances of "should" throughout to what will or will not happen.

File: pkg/alertmanager/multitenant_test.go
Comment (osg-grafana) [reply pkg/alertmanager/multitenant_test.go:2178]: Spell out terms through for content in plain text. Feel free to ignore this nit if you are pressed for time.

File: CHANGELOG.md
Comment (pracucci) [inline CHANGELOG.md:11]: We're used to link the PR number. Can you open a follow up PR to change `#2772` to `#3541`, please?

File: pkg/alertmanager/multitenant.go
Comment (pracucci) [inline pkg/alertmanager/multitenant.go:307]: [nit] Not needed to export it. Could be `computeFallbackConfig()`.

**Comment by pracucci** [comment]
Nice job, thanks!

**Pull request merged**
